packages/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy/lib/LinkQueueLogger.ts

```diff
@@ -0,0 +1,209 @@
+    } else {
+      this.reachabilityRatio[event].unknown += 1;
+    }
+  }
+
+  public override push(link: ILink, parent: ILink): boolean {
+    const resp: boolean = super.push(link, parent);
+    if (resp) {
+      const reachabilityCriteriaParentLink = LinkQueueLogger.getReachability(parent);
+
+      const parentStatisticLink: IURLStatistic = {
+        url: parent.url,
+        reachability_criteria: reachabilityCriteriaParentLink === null ? null : reachabilityCriteriaParentLink[0],
+        reachability_criteria_dynamic_info: reachabilityCriteriaParentLink === null ?
+          undefined :
+          reachabilityCriteriaParentLink[1],
+      };
+
+      const reachabilityCriteriaCurrentLink = LinkQueueLogger.getReachability(link);
+
+      const statisticLink: IURLStatistic = {
+        url: link.url,
+        reachability_criteria: reachabilityCriteriaCurrentLink === null ? null : reachabilityCriteriaCurrentLink[0],
+        reachability_criteria_dynamic_info: reachabilityCriteriaCurrentLink === null ?
+          undefined :
+          reachabilityCriteriaCurrentLink[1],
+        timestamp: Date.now(),
+        parent: parentStatisticLink,
+      };
+      this.appendReachabilityStatistic(statisticLink, 'pushEvent');
+      const event: ILinkQueueEvent = {
+        type: EventType.Push,
+        link: statisticLink,
+        query: this.query,
+        queueStatistics: {
+          size: this.getSize(),
+          reachabilityRatio: this.reachabilityRatio,
+        },
+      };
+
+      this.materialize(event);
+    }
+    return resp;
+  }
+
+  public override pop(): ILink | undefined {
+    const link = super.pop();
+    if (link !== undefined) {
+      const reachabilityCriteriaCurrentLink = LinkQueueLogger.getReachability(link);
+      const statisticLink: IURLStatistic = {
+        url: link.url,
+        reachability_criteria: reachabilityCriteriaCurrentLink === null ? null : reachabilityCriteriaCurrentLink[0],
+        reachability_criteria_dynamic_info: reachabilityCriteriaCurrentLink === null ?
+          undefined :
+          reachabilityCriteriaCurrentLink[1],
+        timestamp: Date.now(),
+      };
+      this.appendReachabilityStatistic(statisticLink, 'popEvent');
+      const event: ILinkQueueEvent = {
+        type: EventType.Pop,
+        link: statisticLink,
+        query: this.query,
+        queueStatistics: {
+          size: this.getSize(),
+          reachabilityRatio: this.reachabilityRatio,
+        },
+      };
+
+      this.materialize(event);
+    }
+    return link;
+  }
+
+  /**
+   * Materialize the current event into the logger
+   * @param {ILinkQueueEvent} event - Current event
+   */
+  private materialize(event: ILinkQueueEvent): void {
+    const jsonEvent = { ...event, type: EventType[event.type] };
+    this.logger.trace(LinkQueueLogger.LINK_QUEUE_EVENT_NAME, { data: JSON.stringify(jsonEvent) });
+  }
+}
+
+/**
+ * The type of event
+ */
+export enum EventType {
+  Push,
```

Comment:
I'm surprised the linter doesn't signal this, but enum values are usually all UPPERCASE.